feat: port to Panacea

youngjoon-lee opened the pull request:
### Changes
- Changed an app name from `Cosmos` to `MediBloc`.
- Changed a coin type in the HD path from `118` to `371`.
- Changed a default Bech32 prefix from `cosmos` to `panacea`
- Changed app icons.
- Changed hardcoded values in unit tests.
   - Also, changed app screenshots for zemu tests

### Release plan
- Will be released as `v2.34.2-panacea` because this PR is a fork of `cosmos/ledger-cosmos@v2.34.2`.

cl9200 reviewed (approved):
:+1:

Commit:
feat: port to Panacea

gyuguen commented on docs/TXSPEC.md:
```diff
@@ -26,9 +26,9 @@ Transactions passed to the Ledger device will be in the following format. The Le
+  "chain_id": "panacea-3",
   "fee": {
-    "amount": [{"amount": "4000", "denom": "uatom"}, ...],
+    "amount": [{"amount": "4000", "denom": "umed"}, ...],
     "gas": "40000"
```
It seems that our chain has set a fee of 5umed and 200000 gas.

youngjoon-lee replied:
This document is only for an example. The setting of 5umed and 200000gas is configured in each network. So, this ledger implementation doesn't need to care about those settings.

youngjoon-lee replied:
Specifically, the ledger app always uses configurations that are specified by `--fees`, `--gas` and `--gas-price`.

gyuguen replied:
I got it. Thank you

gyuguen commented on tests/json_parser.cpp:
```diff
@@ -365,7 +365,7 @@ namespace {
     TEST(TxValidationTest, ObjectGetValueCorrectFormat) {
         auto transaction =
-                R"({"account_number":"0","chain_id":"test-chain-1","fee":{"amount":[{"amount":"5","denom":"photon"}],"gas":"10000"},"memo":"testmemo","msgs":[{"inputs":[{"address":"cosmosaccaddr1d9h8qat5e4ehc5","coins":[{"amount":"10","denom":"atom"}]}],"outputs":[{"address":"cosmosaccaddr1da6hgur4wse3jx32","coins":[{"amount":"10","denom":"atom"}]}]}],"sequence":"1"})";
+                R"({"account_number":"0","chain_id":"test-chain-1","fee":{"amount":[{"amount":"5","denom":"photon"}],"gas":"10000"},"memo":"testmemo","msgs":[{"inputs":[{"address":"panaceaaccaddr1d9h8qat5e4ehc5","coins":[{"amount":"10","denom":"atom"}]}],"outputs":[{"address":"panaceaaccaddr1da6hgur4wse3jx32","coins":[{"amount":"10","denom":"atom"}]}]}],"sequence":"1"})";
```
It seems that you need to change the atom to med here.

youngjoon-lee replied:
Actually, it doesn't matter when considering the way how this test works, but I will change it for clarity.

gyuguen commented on tests/testcases/manual.json:
```diff
@@ -595,12 +595,12 @@
       "fee": {
         "amount": [
           {
             "amount": "5000",
```
So is this
```suggestion
            "amount": "5",
```

youngjoon-lee replied:
We don't need to change this value.
https://github.com/medibloc/ledger-panacea/pull/1#discussion_r853955886

youngjoon-lee replied:
Actually, we also don't need to change the chain ID from cosmoshub-4 to panacea-3.

gyuguen commented on tests/tx_parse.cpp:
```diff
@@ -141,7 +141,7 @@ namespace {
 
     TEST(TxParse, Tx_Page_Count) {
-        auto transaction = R"({"account_number":"0","chain_id":"test-chain-1","fee":{"amount":[{"amount":"5","denom":"photon"}],"gas":"10000"},"memo":"testmemo","msgs":[{"inputs":[{"address":"cosmosaccaddr1d9h8qat5e4ehc5","coins":[{"amount":"10","denom":"atom"}]}],"outputs":[{"address":"cosmosaccaddr1da6hgur4wse3jx32","coins":[{"amount":"10","denom":"atom"}]}]}],"sequence":"1"})";
+        auto transaction = R"({"account_number":"0","chain_id":"test-chain-1","fee":{"amount":[{"amount":"5","denom":"photon"}],"gas":"10000"},"memo":"testmemo","msgs":[{"inputs":[{"address":"panaceaaccaddr1d9h8qat5e4ehc5","coins":[{"amount":"10","denom":"atom"}]}],"outputs":[{"address":"panaceaaccaddr1da6hgur4wse3jx32","coins":[{"amount":"10","denom":"atom"}]}]}],"sequence":"1"})";
```
It is also marked as an atom here, is there any problem?

gyuguen reviewed (approved):
Thank you for your hard work.

Commit:
change app name to MediBloc

Commit:
update icons

youngjoon-lee commented:
@gyuguen @hansol-medi 
Changed the app name from `Panacea` to `MediBloc`: [91bf466](https://github.com/medibloc/ledger-panacea/pull/1/commits/91bf46612a021d10ab7f1eaf70bad2425c91e988)
Changed app icons: [71db396](https://github.com/medibloc/ledger-panacea/pull/1/commits/71db3962df0fceaa9557aaa1decde759012ba256)

0xHansLee reviewed (approved):
Thank you for your efforts!! 
We have our ledger app 🙌
